File: daily/Backtracking/78-Subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        results = []
        def backtrack(start, path):
            results.append(path[:])
            for i in range(start, len(nums)):
                path.append(nums[i])
                backtrack(i + 1, path)
                path.pop()

        backtrack(0, [])
        return results

File: daily/Backtracking/78-Subsets.go
package daily

func subsets(nums []int) [][]int {
	var results [][]int
	var path []int

	var backtrack func(start int)
	backtrack = func(start int) {
		subset := make([]int, len(path))
		copy(subset, path)
		results = append(results, subset)

		for i := start; i < len(nums); i++ {
			path = append(path, nums[i])
			backtrack(i + 1)
			path = path[:len(path)-1]
		}
	}
	backtrack(0)
	return results
}
